### PR TITLE
Adding Contrail SDN to OpenShift

### PR DESCRIPTION
**Pull request by savithruml**
(Closed) Previous PR: Juniper Contrail SDN support for OpenShift [#7514](https://github.com/openshift/openshift-ansible/pull/7514)

Addresses:

- Changes @rajatchopra had asked for

### CODE REVIEW

File: roles/contrail_node/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+- name: Stat for Contrail docker images
+  stat:
+    path: "/tmp/{{ item }}-{{ contrail_os_release }}-{{ contrail_version }}.tar.gz"
+  register: image
+  with_items:
+  - contrail-agent
+  - contrail-kubernetes-agent
+
+- name: Stat for loopback plugin
```
Comment (rajatchopra) [inline roles/contrail_node/tasks/main.yaml]: Can't the daemonset image have the loopback pre-installed in it?

File: roles/contrail_node/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      state: directory
+      
+  - name: Download & extract loopback CNI binary
+    unarchive:
+      src: "https://github.com/containernetworking/cni/releases/download/{{ cni_version }}/cni-{{ cni_version }}.tgz"
+      dest: "/opt/cni/bin"
+      mode: a+x
+      remote_src: True
+
+- name: Add Contrail service account to privileged scc
```
Comment (rajatchopra) [inline roles/contrail_node/tasks/main.yaml]: Why do we use delegate_to: contrail_master and run it in the contrail_node tasks? We could just run this with master's tasks, right?

File: playbooks/adhoc/uninstall.yml [hunk 3]
```diff
@@ -574,3 +597,19 @@
...
+  become: yes
+  tasks:
+  - name: Find nodes which has Contrail vRouter
+    when: item == "vhost0"
+    with_items:
+      - "{{ ansible_interfaces | map('replace', '-','_') | list }}"
+    register: vhost_register
+
+  - name: Reboot Contrail vRouter nodes
+    shell: /sbin/shutdown -r +1
```
Comment (rajatchopra) [inline playbooks/adhoc/uninstall.yml]: I thought this was all daemonset. shutdown?

File: roles/contrail_node/tasks/main.yaml (hunk 1 above)
Comment (savithruml) [reply roles/contrail_node/tasks/main.yaml]: Hi @rajatchopra we don't package that & pull from upstream

File: roles/contrail_node/tasks/main.yaml (hunk 2 above)
Comment (savithruml) [reply roles/contrail_node/tasks/main.yaml]: These tasks are post "contrail_master" plays

File: playbooks/adhoc/uninstall.yml (hunk 3 above)
Comment (savithruml) [reply playbooks/adhoc/uninstall.yml]: Reboot the vRouter nodes after cleaning up the vRouter processes

Comment (rajatchopra) [reply playbooks/adhoc/uninstall.yml]: why is a reboot needed?

Comment (rajatchopra) [reply playbooks/adhoc/uninstall.yml]: Aren't these nodes where there could be other pods running too?

**Comment by vrutkovs** [comment]
Master branch is closed! A major refactor is ongoing in devel-40. Changes for 3.x should be made directly to the latest release branch they're relevant to and backported from there.

**Pull request closed without merging**